### PR TITLE
DT-950: Handle NPE in delete study API

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DT-950

### Summary
The DELETE study api is expecting that all studies have datasets although this is not enforced in the code. This PR allows the delete to continue if there are no datasets in the study. It also adds tests to complete the coverage of this endpoint.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by rjohanek — APPROVED** [review]
looks good. do we need a test for the case where datasetIds is null too?

File: src/main/java/org/broadinstitute/consent/http/resources/StudyResource.java [hunk 1]
```diff
@@ -148,7 +148,7 @@ public Response deleteStudyById(@Auth AuthUser authUser, @PathParam("studyId") I
         throw new NotFoundException("Study not found");
       }
 
-      boolean deletable = study.getDatasets()
+      boolean deletable = (study.getDatasets() == null || study.getDatasets().isEmpty()) || study.getDatasets()
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/consent/http/resources/StudyResource.java:151]: Just double checking the intent here: if the study is null or empty, then deletable is true (implying that the study is deletable). Is this correct?

**Review by fboulnois — APPROVED** [review]
Looks good 👍

**Comment by rushtong** [comment]
> looks good. do we need a test for the case where datasetIds is null too?

Good question @raejohanek, thanks. I added a null check test which uncovered another potential NPE on dataset ids, which is also now handled.
